x-pack/plugins/ml/public/components/controls/select_interval/select_interval.js

```diff
@@ -7,65 +7,73 @@
 
 
 /*
- * AngularJS directive for rendering a select element with various interval levels.
+ * React component for rendering a select element with various aggregation interval levels.
  */
-
 import _ from 'lodash';
+import React, { Component } from 'react';
 
-import { stateFactoryProvider } from 'plugins/ml/factories/state_factory';
+import {
+  EuiSelect
+} from '@elastic/eui';
 
-import template from './select_interval.html';
-import 'plugins/ml/components/controls/controls_select';
 
-import { uiModules } from 'ui/modules';
-const module = uiModules.get('apps/ml');
+const OPTIONS = [
+  { value: 'auto', text: 'Auto' },
+  { value: 'hour', text: '1 hour' },
+  { value: 'day', text: '1 day' },
+  { value: 'second', text: 'Show all' }
+];
 
-module
-  .service('mlSelectIntervalService', function (Private) {
-    const stateFactory = Private(stateFactoryProvider);
-    this.state = stateFactory('mlSelectInterval', {
-      interval: { display: 'Auto', val: 'auto' }
-    });
-  })
-  .directive('mlSelectInterval', function (mlSelectIntervalService) {
-    return {
-      restrict: 'E',
-      template,
-      link: function (scope, element) {
-        scope.intervalOptions = [
-          { display: 'Auto', val: 'auto' },
-          { display: '1 hour', val: 'hour' },
-          { display: '1 day', val: 'day' },
-          { display: 'Show all', val: 'second' }
-        ];
-
-        const intervalState = mlSelectIntervalService.state.get('interval');
-        const intervalValue = _.get(intervalState, 'val', 'auto');
-        let intervalOption = scope.intervalOptions.find(d => d.val === intervalValue);
-        if (intervalOption === undefined) {
-          // Attempt to set value in URL which doesn't map to one of the options.
-          intervalOption = scope.intervalOptions.find(d => d.val === 'auto');
-        }
-        scope.interval = intervalOption;
-        mlSelectIntervalService.state.set('interval', scope.interval);
-
-        scope.setInterval = function (interval) {
-          if (!_.isEqual(scope.interval, interval)) {
-            scope.interval = interval;
-            mlSelectIntervalService.state.set('interval', scope.interval).changed();
-          }
-        };
-
-        function setScopeInterval() {
-          scope.setInterval(mlSelectIntervalService.state.get('interval'));
-        }
-
-        mlSelectIntervalService.state.watch(setScopeInterval);
-
-        element.on('$destroy', () => {
-          mlSelectIntervalService.state.unwatch(setScopeInterval);
-          scope.$destroy();
-        });
-      }
+function optionValueToInterval(value) {
+  // Builds the corresponding interval object with the required display and val properties
+  // from the specified value.
+  const option = OPTIONS.find(opt => opt.value === value);
```

Comment:
nit. conditions should be wrapped in braces.
`const option = OPTIONS.find(opt => (opt.value === value));`